### PR TITLE
add `MatchingRules` to `SearchStreamOutput` & `EditHistoryTweet_IDs` to `Tweet`

### PR DESCRIPTION
**Pull request by cryptodeal**
Have a use case where I need the `matching_rules` returned as part of the data when streaming tweets. (Per [twitter API docs](https://developer.twitter.com/en/docs/twitter-api/tweets/filtered-stream/api-reference/get-tweets-search-stream), `matching_rules` is a default field in the data returned by the stream)

### CODE REVIEW

**Comment by cryptodeal** [comment]
Updated so all default fields returned by Twitter API are returned (previously missing `edit_history_tweet_ids` and when streaming tweets, `matching_rules`).

**Comment by michimani** [comment]
@cryptodeal  
Thanks for your contribution!

The general outline of the changes are fine, but there are a few points we would like you to correct.

File: resources/tweet.go [hunk 1]
```diff
@@ -3,26 +3,27 @@ package resources
...
-	PossiblySensitive  *bool               `json:"possibly_sensitive,omitempty"`
-	PromotedMetrics    *PromotedMetrics    `json:"promoted_metrics,omitempty"`
-	PublicMetrics      *TweetPublicMetrics `json:"public_metrics,omitempty"`
-	ReferencedTweets   []ReferencedTweet   `json:"referenced_tweets,omitempty"`
-	ReplySettings      *string             `json:"reply_settings,omitempty"`
-	Source             *string             `json:"source,omitempty"`
-	Withheld           *TweetWithheld      `json:"withheld,omitempty"`
+	ID                     *string             `json:"id"`
+	Text                   *string             `json:"text"`
+	Edit_History_Tweet_Ids []*string           `json:"edit_history_tweet_ids"`
```
Comment (michimani) [inline resources/tweet.go:8]: The field name should be UpperCamelCase please. ( `EditHistoryTweetIDs`)

File: tweet/filteredstream/types/response.go [hunk 2]
```diff
@@ -41,7 +46,8 @@ type SearchStreamOutput struct {
 		Media  []resources.Media `json:"media,omitempty"`
 		Polls  []resources.Poll  `json:"polls,omitempty"`
 	} `json:"includes,omitempty"`
-	Errors []resources.PartialError `json:"errors,omitempty"`
+	Matching_Rules []SearchStreamMatchedRule `json:"matching_rules,omitempty"`
```
Comment (michimani) [inline tweet/filteredstream/types/response.go:49]: The field name should be UpperCamelCase please. (`MatchingRules`)

File: tweet/filteredstream/types/response.go [hunk 3]
```diff
@@ -32,6 +32,11 @@ func (r *DeleteRulesOutput) HasPartialError() bool {
 	return !(r.Errors == nil || len(r.Errors) == 0)
 }
 
+type SearchStreamMatchedRule struct {
+	Tag *string `json:"tag"`
+	Id  *string `json:"id"`
```
Comment (michimani) [inline tweet/filteredstream/types/response.go:37]: The field name should be `ID` please.

**Comment by michimani** [comment]
That is all I would like to see corrected. Thank you in advance.

**Comment by cryptodeal** [comment]
No problem! My bad on the initially incorrect naming conventions, but should be all fixed now!

**Comment by michimani** [comment]
Thanks for the quick fix!

**Comment by michimani** [comment]
I will merge this pull request.

Again, thank you for your contribution! @cryptodeal

**Review by michimani — APPROVED** [review]

**Pull request merged**